llvm/lib/IR/StructuralHash.cpp

```diff
@@ -43,7 +43,7 @@ class StructuralHashImpl {
   }
 
 public:
-  StructuralHashImpl() : Hash(4) {}
+  StructuralHashImpl() = default;
```

Comment:
What's the reason for marking the constructor as default here instead of just omitting it?

Reply:
AFAIK, under `public:`, it's for the readers, not the generated code.  Providing the defaulted default constructor tells the readers that the defaulted default constructor is available (provided that the code compiles with `-Werror`).  If the class contained a member that is not default constructible, then an attempt to declare the defaulted default constructor would trigger a warning (see https://godbolt.org/z/7vTGYb6Pn).